Commons/src/main/java/eu/neclab/ngsildbroker/commons/storage/StorageDAO.java

```diff
@@ -477,4 +482,21 @@ protected String getTenant(String tenantId){
 		}
 		return tenantId;
 	}
+	
+	private String validateDataBaseNameByTenantId(String tenantid) {
+		if (tenantid == null)
+			return null;
+		try {
+			String databasename = "ngb" + tenantid;
```

Comment:
I tried to implement this logic already but it was not working in this case. As @postcontruct calls only one time so new tenant gets not added in tenant2Templates so we have to call the query.

Reply:
Hi, new tenants are added in line 121 so this gets filled over time when someone creates an entity or so.
Just put the check before you query the database and if it's in tenant2Templates we can already return here.
If not we still need to do the sql check you implemented to be sure that it is not there (there could be multiple instances of scorpio writing and the one querying maybe hasn't created a template yet).
In principal on a single instance flow would be something like this: 
Someone queries for tenant a and it's not there yet. so you check in tenant2Templates and in the DB and return tenant not found.
Someone creates an entity in tenant a. this will go through the getTemplates method and add the new template in line 121.
Now someone queries again for tenant a and tenant a is than in the tenant2Templates map already so no need to validate this again via the database.

Reply:
Hi, I have done the recommended changes in the code and validated it.